packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -325,17 +324,15 @@ class BoxDecoration extends Decoration {
   }
 
   @override
-  int get hashCode {
-    return hashValues(
+  int get hashCode => Object.hash(
       color,
       image,
       border,
       borderRadius,
-      hashList(boxShadow),
+      Object.hashAll(boxShadow ?? <Object>[]),
```

Comment:
Is this the correct way to hash a nullable list?
Follow-up, should I use `<Object>[]` or `List<Object>.empty()`?

Reply:
looks equivalent to what we had before, though use `const <Object>[]` to avoid allocating a new list